README.md

```diff
@@ -11,6 +11,39 @@ yarn add unionize
 ## Example
 
 Provide `unionize` with a mapping of tags to value types:
+
+```ts
+  import { unionize, ofType } from 'unionize';
+
+  const Actions = unionize({
+    ADD_TODO: ofType<{ id: string; text: string }>(),
+    CLEAR_TODOS: {}, // For "empty" types, just use {}
+  });
+```
+
+Extract the inferred tagged union:
+
+```ts
+import { UnionOf } from 'unionize'
+
+type Action = UnionOf<typeof Actions>;
+```
+
+The inferred type is:
+
+```ts
+type Action = ({
+    tag: "ADD_TODO";
+} & {
+    id: string;
+    text: string;
+}) | {
+    tag: "CLEAR_TODOS";
+}
+```
+
+We can also optionally override the tag and/or value property names: 
```

Comment:
Since before this point there was no notion of a "value" property, it doesn't really make sense to talk about overriding it. Maybe just

```suggestion
We can also optionally provide tag and/or value property names:
```